### PR TITLE
xds: support getting logical DNS and aggregate cluster configurations from CDS responses

### PR DESCRIPTION
**Pull request by voidzcy**
~CIs should pass after #7685 as it depends on policy/config definition of ClusterResolverLoadBalancer.~

I am splitting out the part for migrating to ClusterResolverLoadBalancer as the child LB for CdsLoadBalancer to a separate PR. This PR should only change how the XdsClient parses CDS responses. It can be reviewed and merged separately. No dependency needed from #7685.

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/XdsClient.java [hunk 1]
```diff
@@ -193,93 +193,19 @@ public boolean equals(Object o) {
   }
 
   static final class CdsUpdate implements ResourceUpdate {
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/XdsClient.java:195]: TODO in a separate PR: I may want to clean up other resource update data structures as well, they are too verbose to use.

File: xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java [hunk 2]
```diff
@@ -193,36 +194,37 @@ private CdsLbState() {
     }
 
     @Override
-    public void onChanged(final CdsUpdate newUpdate) {
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java:196]: Name too verbose, causing line-to-long issue.

File: xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java [hunk 3]
```diff
@@ -193,36 +194,37 @@ private CdsLbState() {
...
+    public void onChanged(final CdsUpdate update) {
       syncContext.execute(new Runnable() {
         @Override
         public void run() {
           if (shutdown) {
             return;
           }
-          if (logger.isLoggable(XdsLogLevel.INFO)) {
-            logger.log(XdsLogLevel.INFO, "Received cluster update from xDS client {0}: "
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java:204]: Log too verbose.

File: xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java [hunk 4]
```diff
@@ -193,36 +194,37 @@ private CdsLbState() {
...
-                    + "cluster_name={1}, eds_service_name={2}, lb_policy={3}, report_load={4}",
-                xdsClient, newUpdate.getClusterName(), newUpdate.getEdsServiceName(),
-                newUpdate.getLbPolicy(), newUpdate.getLrsServerName() != null);
+          // TODO(chengyuanzhang): implementations for logical DNS and aggregate clusters.
+          if (update.clusterType != ClusterType.EDS) {
+            logger.log(XdsLogLevel.WARNING, "Unsupported cluster type: {0}", update.clusterType);
+            return;
           }
-          // FIXME(chengyuanzhang): handle error correctly to avoid being unnecessarily fragile.
-          checkArgument(newUpdate.getLbPolicy().equals("round_robin"),
```
Comment (voidzcy) [inline xds/src/main/java/io/grpc/xds/CdsLoadBalancer.java:210]: This check is useless, "round_robin" is hard coded for CdsUpdate.

File: xds/src/main/java/io/grpc/xds/ClientXdsClient.java [hunk 5]
```diff
@@ -70,6 +74,8 @@
   // Longest time to wait, since the subscription to some resource, for concluding its absence.
   @VisibleForTesting
   static final int INITIAL_RESOURCE_FETCH_TIMEOUT_SEC = 15;
+  @VisibleForTesting
+  static final String AGGREGATE_CLUSTER_TYPE_NAME = "envoy.cluster.aggregate";
```
Comment (dapengzhang0) [inline xds/src/main/java/io/grpc/xds/ClientXdsClient.java:78]: .cluster. or .cluster**s**.?

File: xds/src/main/java/io/grpc/xds/ClientXdsClient.java [hunk 6]
```diff
@@ -280,88 +290,42 @@ protected void handleCdsResponse(String versionInfo, List<Any> resources, String
...
-      // LRS load reporting will be disabled.
-      if (cluster.hasLrsServer()) {
-        if (!cluster.getLrsServer().hasSelf()) {
-          errorMessage = "Cluster " + clusterName + " : only support enabling LRS for the same "
-              + "management server.";
+      String lbPolicy = "round_robin";
+      CdsUpdate update = null;
+      switch (cluster.getClusterDiscoveryTypeCase()) {
+        case TYPE:
+          update = parseNonAggregateCluster(cluster, nonce, lbPolicy, edsResources);
```
[the remark this answers is not in the document]
Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/ClientXdsClient.java:315]: Oh, it not big difference call it right after switch. I had misimpression that the check looked a lot after here in the unified diff view.

**Review by dapengzhang0 — APPROVED** [review]
The flow LGTM. Make sure the integration not breaking.

File: xds/src/main/java/io/grpc/xds/ClientXdsClient.java (hunk 5 above)
Comment (voidzcy) [reply xds/src/main/java/io/grpc/xds/ClientXdsClient.java]: Oops, that's an amazing catch. Thanks 😄

**Pull request merged**